Branch-SDK/src/main/java/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java

```diff
@@ -81,17 +76,20 @@ private BranchResponse doRestfulGet(String url, int retryNumber) throws BranchRe
                 retryNumber++;
                 return doRestfulGet(url, retryNumber);
             } else {
+                BranchResponse result;
                 try {
                     if (responseCode != HttpsURLConnection.HTTP_OK && connection.getErrorStream() != null) {
-                        return new BranchResponse(getResponseString(connection.getErrorStream()), responseCode, Strings.emptyToNull(requestId));
+                        result = new BranchResponse(getResponseString(connection.getErrorStream()), responseCode);
                     } else {
-                        return new BranchResponse(getResponseString(connection.getInputStream()), responseCode, Strings.emptyToNull(requestId));
+                        result = new BranchResponse(getResponseString(connection.getInputStream()), responseCode);
                     }
                 } catch (FileNotFoundException ex) {
                     // In case of Resource conflict getInputStream will throw FileNotFoundException. Handle it here in order to send the right status code
                     PrefHelper.Debug("A resource conflict occurred with this request " + url);
-                    return new BranchResponse(null, responseCode, Strings.emptyToNull(requestId));
+                    result = new BranchResponse(null, responseCode);
                 }
+                result.requestId = Strings.emptyToNull(requestId);
```

Comment:
I removed `requestId` from `BranchResponse` because it's a public class that clients sometimes need to implement, e.g. https://github.com/BranchMetrics/android-branch-deep-linking-attribution/issues/605#issuecomment-431330448